### ISSUE TITLE
TypeVariableImpl cannot be cast to java.lang.Class

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#4]
```
What steps will reproduce the problem?
1. Run attached class' main method

What is the expected output? What do you see instead?


What version of the product are you using? On what operating system?
1.0

Please provide any additional information below.

Hi,

there is some problem with generic type collections. Is it possible to fix 
this? In case such generic types Orika tries to find class with generic name, 
and in case <P extends Person> parameter, there is no 'P' class.


```

Original issue reported on code.google.com by `mathew...@gmail.com` on 13 Feb 2012 at 2:39

Attachments:
- [Mapper.java](https://storage.googleapis.com/google-code-attachments/orika/issue-4/comment-0/Mapper.java)

**Comment by GoogleCodeExporter** [comment]
```
We're working on the issue we will release it soon
```

Original comment by `elaat...@gmail.com` on 26 Feb 2012 at 1:03
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
This should now be fixed with the 1.1-generics branch (soon to be merged into 
trunk).
Take a look at ma.glasnost.orika.test.generics.GenericCollectionsTestCase which 
contains your provided test, along with example of how to map it using the 
newly provided generics-support methods.
```

Original comment by `matt.deb...@gmail.com` on 7 Mar 2012 at 10:47
- Changed state: **Fixed**
